index.ts

```diff
@@ -1,82 +1,209 @@
+      attachments = result.items;
+    },
+  );
+
+  let isAskedToExit = false;
+
   console.log("************* Commands *************");
   console.log("--insert my new task");
   console.log("   Inserts a task");
-  console.log("   Example: \"--insert Get Milk\"");
+  console.log('   Example: "--insert Get Milk"');
   console.log("--toggle myTaskTd");
   console.log("   Toggles the isComplete property to the opposite value");
-  console.log("   Example: \"--toggle 1234abc\"");
+  console.log('   Example: "--toggle 1234abc"');
   console.log("--delete myTaskTd");
   console.log("   Deletes a task");
-  console.log("   Example: \"--delete 1234abc\"");
+  console.log('   Example: "--delete 1234abc"');
   console.log("--list");
   console.log("   List the current tasks");
+  console.log("--add path_to_attachment");
+  console.log("   adds an attachment from the path provided.");
+  console.log('   Example: "--add ../files/image.png"');
+  console.log("--attachments");
+  console.log("   List the current attachments");
   console.log("--exit");
   console.log("   Exits the program");
   console.log("************* Commands *************");
 
   const rl = readline.createInterface({ input, output });
+
   while (!isAskedToExit) {
+    let answer = await rl.question("Your command:");
 
-      let answer = await rl.question('Your command:')
-      if (answer.startsWith("--insert")) {
-        let body = answer.replace("--insert ", "")
-        ditto.store.collection("tasks").upsert({
-          body,
-          isDeleted: false,
-          isCompleted: false
-        })
-      }
-      if (answer.startsWith("--toggle")) {
-        let id = answer.replace("--toggle ", "")
-        ditto.store.collection("tasks")
-        .findByID(id).update((doc) => {
-          let isCompleted = doc.value.isCompleted
-          doc.at("isCompleted").set(!isCompleted)
-        })
-      }
-      if (answer.startsWith("--list")) {
-        console.log(tasks.map((task) => task.value))
-      }
-      if (answer.startsWith("--delete")) {
-        let id = answer.replace("--delete ", "")
-        ditto.store.collection("tasks")
-        .findByID(id).update((doc) => {
-          doc.at("isDeleted").set(true)
-        })
-      }
-      if (answer.startsWith("--exit")) {
-        ditto.stopSync()
-        process.exit()
-        
-      }
-  }
+    if (answer.startsWith("--insert")) {
+      let body = answer.replace("--insert ", "");
+      const newTask = { body, isDeleted: false, isCompleted: false };
+
+      await ditto.store.execute(
+        `
+        INSERT INTO tasks
+        DOCUMENTS (:newTask)
+        `,
+        {
+          newTask,
+        },
+      );
+    }
+
+    if (answer.startsWith("--add")) {
+      let path = answer.replace("--add ", "");
+      const metadata = { name: "image.png" };
+
+      // Copy the file into Ditto's store and create an attachment object.
+      const myAttachment = await ditto.store.newAttachment(path, metadata);
```

Comment:
needs some error handling probably, at least for code `store/attachment-file-not-found`